src/server/users/models.js

```diff
@@ -179,46 +179,84 @@ export class CreatedPostManager {
  *    state: 'alive',
  *    key: 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
  *    userKey: 'user-8000000000000-uuiduuid-uuid-uuid-uuid-uuiduuiduuid',
- *    entity: 'image',
- *    imageUrl: 'url-of-image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000'
+ *    isCreatedByUser: false,
+ *    imageUrls: ['url-of-image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000']
  * }
 **/
-  static getPost({entity, itemKey, imageKey}, cb) {
-    db.get(itemKey, (err, item) => {
-      if (err) {
-        return cb(err);
+  static getPost({isCreatedByUser, itemKey, imageKeys}) {
+    return getPromise(itemKey).then((item)=>{
+      item.isCreatedByUser = isCreatedByUser;
+      item.imageUrls = new S3Connector().getImageUrls(imageKeys);
+      return item;
+    });
+  }
+/**
+ * combine key sets for unique item key
+ * @param {Array} keySets example:
+ * [
+ *    {  entity  : 'image',
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKey: 'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000'
+ *    }
+ *    {  entity  : 'item',
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKey: 'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b999'
+ *    }
+ * ]
+ * @return {Array} item example:
+ * [
+ *    {  isCreatedByUser : true,
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKeys :
+ *        [
+ *          'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *          'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b999'
+ *        ]
+ *    }
+ * ]
+**/
+  static combineKeySets(keySets) {
```

Comment:
How about change name from set to list?

'set' sound like data structure set which has no order and also no map function just like line 219.
combinedKeySet sound like set or dictionay and unordered.

I think list is more clear than set in this case